### ISSUE TITLE
EC only texture misalignment.

### ISSUE THREAD

**Issue by LoboEire** [edge-classic/EDGE-classic#767]
In EC, map01 of this wad has a misaligned texture.
[https://www.doomworld.com/forum/topic/128999-squenched-avacausage-bite-1a-boom-map-1-to-13/](https://www.doomworld.com/forum/topic/128999-squenched-avacausage-bite-1a-boom-map-1-to-13/)
It should look like this:
![Screenshot_Doom_20241214_104317](https://github.com/user-attachments/assets/89e500d9-06cb-443f-beee-90eeab346af0)

But in EC it looks like this:
![shot39](https://github.com/user-attachments/assets/6fba882d-efdf-4b77-a6e3-d4503403c19d)

**Comment by pbdot** [comment]
I've been looking into this and it isn't map related.  The texture in question is 72 high, and the offset being applied is 72, normally that would just wrap back to 0.  However, non-power-of-two textures are promoted, so in this case becomes 128 high...  and then have their margins filled.  If you don't fill the margin you can see below where the offset is landing things... ie.  in the non-tiling "filled" space

I looked into how DSDA doom handles this and it checks whether a surface needs to tile based on the height, though that doesn't support the scaling and stuff that EC does.  It is possible to fix this for this specific case, by checking the texture height against the offset, though would really like a more holistic fix. 

Basically, you can't get a 72 px high texture to ever tile in a power of two, and the offsets are being applied in the 72 pixel space, when the actual texture is 128 high. 

![shot13](https://github.com/user-attachments/assets/932ec457-efa6-4bb8-b7df-c020fdce3af0)

**Comment by LoboEire** [comment]
Thanks for the explanation.
Seems like a potential fix would need careful consideration or it could cause more issues with other things than it's worth.

**Comment by pbdot** [comment]
It is feasible I think to fix this, though going to move off the 1.5 milestone
